shepherd_bms/include/segment.h

```diff
@@ -1,7 +1,115 @@
 #ifndef SEGMENT_H
 #define SEGMENT_H
 
+#include <LTC68041.h>
+#include <nerduino.h>
 #include "datastructs.h"
 
+#define NUM_SEGMENTS    4
+#define NUM_CHIPS       NUM_SEGMENTS*2
+
+#define THERM_REF       4700            // Reference resistor for thermistor measurments / voltage divider
+#define MIN_VOLT        2.9
+#define MAX_VOLT        4.2
+#define MAX_DELTA_V     0.02
+#define BAL_MIN_V       4.00
+
+
+/**
+ * @brief This class serves as the interface for all of the segment boards
+ */
+class SegmentInterface
+{
+    private:
+        const uint32_t TEMP_CONV[52] = 
+        {
+            33200, 31500, 29900, 28400, 26900, 25600, 24300, 23100, 21900, 20900, 
+            19900, 18900, 18000, 17100, 16300, 15500, 14800, 14100, 13500, 12900, 
+            12300, 11700, 11200, 10700, 10200, 9780, 9350, 8940, 8560, 8190, 7840, 
+            7510, 7190, 6890, 6610, 6340, 6080, 5830, 5590, 5370, 5150, 4950, 4750, 
+            4570, 4390, 4220, 4060, 3900, 3750, 3610, 3470, 0
+        };
+
+        const uint32_t VOLT_TEMP_CONV[57] = 
```

Comment:
I think only this voltage based lookup table is used. The TEMP_CONV was an earlier version using resistance and another calculation function